packages/flutter/test/material/tooltip_test.dart

```diff
@@ -1026,7 +1021,12 @@ void main() {
     gesture = null;
   });
 
+
   testWidgets('Calling ensureTooltipVisible on an unmounted TooltipState returns false', (WidgetTester tester) async {
+    // Using [testWidgets] instead of [testWidgetsWithLeakTracking] because
+    // [tooltipState] is hold after disposal by the test
+    // and thus is not garbage collected.
```

Comment:
This makes me question that turning on leak tracking by default is actually feasible. This is a fairly common pattern. If people are confronted with a non-actionable leak warnings every time they obtain a state in a test, they'll likely get frustrated with these warnings and stop paying attention to the legitimate ones.

Reply:
Good question. 
We can add allowlist for such leaks and suggest the code to copy-paste in error message.

Reply:
We talked about this a bit offline but unfortunately without @goderbauer - this test is a _little_ strange in that it's poking at a disposed object after the object has been disposed.

I think a reasonable API would be to somehow add that object as "intentionally leaked", e.g. `tester.ignoredLeakedObject(tooltipState)`. WDYT @goderbauer ?

Reply:
> this test is a little strange in that it's poking at a disposed object after the object has been disposed.

For my own understanding: If you'd remove the last `expect(tooltipState.ensureTooltipVisible(), false);` from the test, the leak tracker would not complain? Or would holding the disposed `tooltipState` in that variable still constitute a leak? I believe, doing the latter is somewhat common...

> I think a reasonable API would be to somehow add that object as "intentionally leaked",

Again, I think to answer this question we need to know how common this situation is, i.e. how often would test authors get confused about a false leak warning? For that, I would check how often this situation happens in our current test code base...

Reply:
I collected some statistics. Out of 93 tests, there is just 1 that holds state after disposal.
Yes, it counts as leak, because an object is expected to be GCed soon after disposal.

Looking forward to discuss details on a meeting.